### PR TITLE
Allow specifying delimiters to access group information in nested claims

### LINKED ISSUE #9: realm_access claim could not be converted to string list
hi,

could you please help me to set `groups_claim` parameter for role? 

I've got error when I tried to set it to `groups_claim=realm_access`

```
* "realm_access" claim could not be converted to string list
```

my token has the following:

```
  "realm_access": {
    "roles": [
      "aaa",
      "bbb"
    ]
  },
```

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #9 

This isn't finished, putting here for the moment to get some early feedback.

### CODE REVIEW

File: path_role.go [hunk 1]
```diff
@@ -322,6 +340,32 @@ func (b *jwtAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical.
...
 }
 
+// parseClaimWithDelimiters parses a given claim string and ensures that we can
+// separate it out into a "map path"
+func parseClaimWithDelimiters(claim, delimiters string) ([]string, error) {
+	if delimiters == "" {
+		return []string{claim}, nil
+	}
+	var ret []string
+	for _, runeVal := range delimiters {
```
Comment (kalafut) [inline path_role.go]: Could you use `strings.IndexAny` in lieu of looping?

**Comment by tyrannosaurus-becks** [comment]
This looks good! It would be great to get some testing and validation from one of the users you mentioned in https://github.com/hashicorp/vault-plugin-auth-jwt/issues/9, I think that's the only thing it needs. However, IMHO, that shouldn't be a blocker if it gets to release time and they haven't had a chance.

File: path_login.go [hunk 2]
```diff
@@ -179,7 +179,32 @@ func (b *jwtAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
...
-		groupsClaimRaw, ok := allClaims[role.GroupsClaim]
+		mapPath, err := parseClaimWithDelimiters(role.GroupsClaim, role.GroupsClaimDelimiterPattern)
+		if err != nil {
+			return logical.ErrorResponse(errwrap.Wrapf("error parsing delimiters for groups claim: {{err}}", err).Error()), nil
+		}
+		if len(mapPath) < 1 {
+			return logical.ErrorResponse("unexpected length 0 of claims path after parsing groups claim against delimiters"), nil
+		}
+		var claimKey string
+		claimMap := allClaims
```
[the remark this answers is not in the document]
Comment (jefferai) [reply path_login.go]: I'm not acutally changing the map, I'm merely drilling down into it for read-only purposes, so I think it should be fine.

**Comment by jefferai** [comment]
No reply yet, merging so it makes it into the beta.

**Pull request merged**